editoast/migrations/2025-07-24-085435_add-round-trips/up.sql

```diff
@@ -0,0 +1,36 @@
+-- This requires the extension `intarray` to be installed.
+CREATE TABLE train_schedule_round_trips (
+    id int8 PRIMARY KEY GENERATED BY DEFAULT AS IDENTITY,
+    left_id INTEGER REFERENCES train_schedule(id) ON DELETE CASCADE NOT NULL UNIQUE,
+    right_id INTEGER REFERENCES train_schedule(id) ON DELETE CASCADE UNIQUE,
+    CONSTRAINT id_order CHECK (
+        right_id IS NULL
+        OR left_id < right_id
+    ),
+    CONSTRAINT train_schedules_unique_round_trips EXCLUDE using gist (
+        (
+            CASE
+                WHEN right_id IS NULL THEN array [left_id]
+                ELSE array [left_id, right_id]
+            END
+        ) WITH &&
```

Comment:
I did not find another way to check this.
I'm not sure about the efficiency of this check. It's also sad that it requires a new extension.